src/Integration.Vsix.UnitTests/Settings/SonarLintSettingsTests.cs

```diff
@@ -22,354 +22,287 @@
 
         [TestMethod]
-        public void IntegrationSettings_Ctor_NullArgChecks()
-        {
-            Exceptions.Expect<ArgumentNullException>(() => new SonarLintSettings((SettingsManager)null));
-        }
+        public void MefCtor_CheckIsSingleton()
+            => MefTestHelpers.CheckIsSingletonMefComponent<SonarLintSettings>();
 
         [TestMethod]
-        public void IntegrationSettings_Ctor_InitializesStore()
+        public void Ctor_DoesNotCallAnyServices()
         {
-            // Sanity - should be empty store
-            this.settingsStore.AssertCollectionDoesNotExist(SonarLintSettings.SettingsRoot);
+            var storeFactory = new Mock<IWritableSettingsStoreFactory>();
+            _ = CreateTestSubject(storeFactory.Object);
 
-            // Act
-            this.CreateTestSubject();
-
-            // Assert
-            this.settingsStore.AssertCollectionExists(SonarLintSettings.SettingsRoot);
+            // The MEF constructor should be free-threaded, which it will be if
+            // it doesn't make any external calls.
+            storeFactory.Invocations.Should().BeEmpty();
         }
 
         [TestMethod]
-        public void IntegrationSettings_GetValueOrDefault_Bool()
+        public void LazyInitialization_FactoryIsOnlyCalledOnce()
         {
-            // Arrange
-            SonarLintSettings testSubject = this.CreateTestSubject();
+            var store = new Mock<WritableSettingsStore>();
+            var factory = CreateStoreFactory(store.Object);
+
+            var testSubject = CreateTestSubject(factory.Object);
 
-            // Test case 1: exists -> value
-            // Arrange
-            bool expected1 = false;
-            this.settingsStore.SetBoolean(SonarLintSettings.SettingsRoot, "key1", expected1);
+            factory.Invocations.Should().BeEmpty(); // Sanity check
+            store.Reset();
 
-            // Act
-            bool actual1 = testSubject.GetValueOrDefault("key1", true);
+            // 1. Make any call to the test subject
+            _ = testSubject.GetValueOrDefault("any", "any");
 
-            // Assert
-            actual1.Should().Be(expected1, "Did not load existing value");
+            factory.Invocations.Should().HaveCount(1);
+            store.Invocations.Should().HaveCount(1);
 
-            // Test case 2: does NOT exist -> default
-            // Arrange
-            bool expected2 = true;
+            // 2. Call the settings again
+            _ = testSubject.GetValueOrDefault("any", true);
+            testSubject.SetValue("any", "any");
 
-            // Act
-            bool actual2 = testSubject.GetValueOrDefault("key2", expected2);
+            factory.Invocations.Should().HaveCount(1);
+            store.Invocations.Should().HaveCount(3);
+        }
 
-            // Assert
-            actual2.Should().Be(expected2, "Did not return default value");
+        #region Boolean method tests
```

Comment:
The interface has `get` and `set` methods for booleans, integers and strings.

There are four tests for each data type:
1. GetValueOrDefault -> value is returned
2. Get when store throws -> default is returned
3. Get when store is null -> default is returned
4. SetValue -> value is set
5. SetValue when store is null -> no error

I couldn't see a simple way to de-duplicated the test code, so I copied and tweaked the tests for each data type.